lib/ansible/module_utils/oneview.py

```diff
@@ -257,7 +257,13 @@ def _check_hpe_oneview_sdk(self):
             self.module.fail_json(msg=self.HPE_ONEVIEW_SDK_REQUIRED)
 
     def _create_oneview_client(self):
-        if not self.module.params['config']:
+        if self.module.params.get('hostname'):
```

Comment:
So I am not sure this is the best way to do this. I would simply look for each of the variables at the various locaions, and merge them all together, using a precedence rule: parameter over config over environment.

But this is something you can add later. Not something you have to worry about now.